subsys/bluetooth/host/hci_core.c

```diff
@@ -7594,6 +7594,12 @@ int bt_le_adv_start_legacy(const struct bt_le_adv_param *param,
 	    (param->options & BT_LE_ADV_OPT_CONNECTABLE)) {
 		err = le_adv_start_add_conn(adv, &conn);
 		if (err) {
+			if (err == -ENOMEM && !dir_adv &&
+			    !(param->options & BT_LE_ADV_OPT_ONE_TIME)) {
+				atomic_set_bit(adv->flags, BT_ADV_PERSIST);
+				return 0;
```

Comment:
At the end of this legacy function a bunch of adv flags including BT_ADV_CONNECTABLE are set from the param options. Don't these flags need to be set in this adv postponement case as well? At least it seems that bt_le_adv_resume will refuse to resume if BT_ADV_CONNECTABLE is not already set in the adv flags.

Reply:
Good catch, thank you.